### ISSUE TITLE
odeint (lsoda) emits too many warnings that are not filterable.

### ISSUE THREAD

**Issue by andyfaff** [scipy/scipy#15940]
### Describe your issue.

I have a colleague who is attempting to use `odeint` to integrate a differential equation. This produces a large volume (>100k lines) of warning messages that, whilst real, they would like to filter as they pollute the Jupyter notebooks and are emitted in the Terminal window, etc.
The same noise is emitted in `solve_ivp` if `method='LSODA'` is used.

We have tried redirecting stdout, stderr, filtering Python warnings, etc, to no success. The message is emitted from [lsoda.f](https://github.com/scipy/scipy/blob/1ab82fffd30f3bf936c1118e55617a978f9b7e90/scipy/integrate/odepack/lsoda.f#L1339), which calls [xerrwv](https://github.com/scipy/scipy/blob/1ab82fffd30f3bf936c1118e55617a978f9b7e90/scipy/integrate/odepack/xerrwv.f#L1).

From reading lsoda.f it should be [possible set the logical unit that the messages are printed to, or to suppress warning messages](https://github.com/scipy/scipy/blob/main/scipy/integrate/odepack/lsoda.f#L798). What does logical unit 6 correspond to in Fortran terms?

My preference at the moment would be to sort out where the messages are directed to, followed by addition of a parameter to the module that would turn off the warning messages.

@llimeht

### Reproducing Code Example

```python
import contextlib
import io
import numpy
import numpy as np
from scipy.integrate import odeint


def dxdt(x, t):
    return [ x[0]**3 - 4. * x[1],
             x[0] - x[1] ]


f = io.StringIO()
with contextlib.redirect_stdout(f), contextlib.redirect_stderr(f):
    t = numpy.linspace(0, 8, 1001)
    x0 = np.array([-1.38918542, -7.87846202])
    x = odeint(dxdt, x0, t)
```

### Error message

```shell
lsoda--  warning..internal t (=r1) and h (=r2) are
       such that in the machine, t + h = t on the next step  
       (h = step size). solver will continue anyway
      in above,  r1 =  0.1750743642151D+00   r2 =  0.1257107638210D-16
<repeats ad nauseam>
```

### SciPy/NumPy/Python version information

1.8.0 1.21.4 sys.version_info(major=3, minor=9, micro=7, releaselevel='final', serial=0)

**Comment by Patol75** [comment]
Although on the one hand, I agree that LSODA warnings could be handled better, on the other hand, you should not get any if your IVP problem is sound and well set up (at least, that is my experience with LSODA).

I updated your code to:
```
import contextlib
import io

import matplotlib.pyplot as plt
from scipy.integrate import solve_ivp


def rhs(t, y):
    return y[0] ** 3 - 4 * y[1], y[0] - y[1]


def jac(t, y):
    return [[3 * y[0] ** 2, -4], [1, -1]]


f = io.StringIO()
with contextlib.redirect_stdout(f), contextlib.redirect_stderr(f):
    sol = solve_ivp(
        rhs,
        [0, 8],
        [-1.38918542, -7.87846202],
        method="LSODA",
        jac=jac,
        min_step=1e-15,
    )

plt.plot(sol.t, sol.y[0])
plt.plot(sol.t, sol.y[1])
plt.show()
```
If I run it, the following figure is generated
![Figure_1](https://user-images.githubusercontent.com/23244688/162356268-fec71e52-a97a-4d58-aafa-4c2dd551d162.png)
and the following single message is printed
```
lsoda--  at t (=r1) and step size h (=r2), the    
       corrector convergence failed repeatedly     
       or with abs(h) = hmin   l
      in above,  r1 =  0.1748361918162D+00   r2 =  0.9696386307155D-15
```

**Comment by laurent90git** [comment]
@Patol75 is right, your solution is simply diverging in a more-than-exponential fashion. Thus, the time step must be decreased to accurately follow the divergence. And at one point the divergence is so fast that the required time step becomes too low, such that when you perform the operation t(n+1) = t(n)+dt, you get t(n+1)=t(n).
The authors of LSODA have chosen to only issue a warning when that occurs.
The other Python integrators from solve_ivp instead raise an Exception. It seems more sensible, because it really indicates that something is wrong with the integration !

**Comment by hbhargava7** [comment]
@andyfaff I agree that this is a serious pain point. The outputs should be filterable. Consider running parameter inference where some parameter regimes may lead to artifactual behavior– a legitimate use case where one would want to ignore the output.
Did you find a way to suppress the output?

**Comment by Jacob-Stevens-Haas** [comment]
Concur on this issue.  The project I work on, github.com/dynamicslab/pysindy, focuses on learning differential equations from data.  Which means that the learned ODE, essentially random, may have singularities and numerical stiffness.  

I run experiments that tweak a lot of paremeters and plot simulation results results.  I want to display stdout and stderr in the results, but I want to be able to catch LSODA warnings myself, print a blurb that says the discovered ODE experiences blow-up and move along to the next set of parameters.  Instead, I get hundreds of these warnings.

**Comment by romanovzky** [comment]
This also bothers me as I am performing these integrations inside a parameter space scan. I've tried many ways to suppress the output without any success.

Looking [here](https://stackoverflow.com/questions/49598051/suppress-odepack-warnings-in-scipy-integrate-ode), it seems that there as an attempt in the past to address this, [here](https://github.com/scipy/scipy/pull/8635). However, this PR suppresses the warning at the fortran level, which is not a fix as the fortran code is not distributed with scipy (at least in my distro it is no in the same scipy site package folder).

**Comment by sanghyukmoon** [comment]
Any update on this? Having the same issue when doing the parameter space scan (precisely, I am solving the ODE inside the root finding and/or optimization problem, so the same ODE is solved multiple times with a range of parameters. It looks like the ODE is ill-posed for some specific combination of parameters, but it doesn't really matter if the final desired parameters yield a sane solution.

**Comment by crisla** [comment]
Same issue here as @sanghyukmoon - I am using stdout_redirected but there has to be a better way.

**Comment by ilayn** [comment]
This will be handled when and if we find an alternative to the F77 code. See #18566 for the pipeline.
